nestedtensor/csrc/functions.cpp

```diff
@@ -219,27 +218,56 @@ NestedTensor interpolate(NestedTensor input,
         throw error;
     }
 
+    TensorNode input_structure = input.get_structure();
     auto options = F::InterpolateFuncOptions().mode(int_mode);
-    if (size.has_value()) {
-      options = options.size(size.value().vec());
-    }
-    
-    if (scale_factor.has_value()) {
-      options = options.scale_factor(scale_factor.value().vec());
-    }
-
     if (align_corners.has_value()) {
       options.align_corners() = align_corners.value();
     }
 
-    TensorNode res = map(
+    // Either scale factor or size can be passed
+    if (scale_factor.has_value()) {
+      options = options.scale_factor(scale_factor.value().vec());
+      TensorNode res = map(
         [&options](at::Tensor input_tensor) {
           return F::interpolate(input_tensor.unsqueeze(0), options).squeeze(0);
         },
         input_structure);
+      return NestedTensor(std::move(res));
+    }
 
-    return NestedTensor(std::move(res));
-}
+    // Get input leaves count
+    auto fn = [](at::Tensor leaf, int64_t input) {
+      return input + 1;
+    };
+    auto leaves_count = reduce<decltype(fn), int64_t, at::Tensor>(input.get_structure(), fn, 0);
+    
+    if (size.has_value()) {
+      if (size.value().size() != 1 && size.value().size() != leaves_count) {
+        std::stringstream error;
+        error << "Interpolate has to take either 1 size tuple or same amount as leaves in Nested Tensor." << std::endl;
```

Comment:
You can use throw std::runtime_error "https://en.cppreference.com/w/cpp/error/runtime_error"